qiita_db/software.py

```diff
@@ -0,0 +1,522 @@
+        r"""Creates a new command in the system
+
+        Parameters
+        ----------
+        software : qiita_db.software.Software
+            The software to which this command belongs to.
+        name : str
+            The name of the command
+        description : str
+            The description of the command
+        cli_cmd : str
+            The CLI used to call this command
+        parameters_table : str
+            The name of the table in which the parameters of the commands are
+            stored
+
+        Returns
+        -------
+        qiita_db.software.Command
+            The newly created command
+        """
+        with TRN:
+            sql = """INSERT INTO qiita.software_command
+                            (name, software_id, description, cli_cmd,
+                             parameters_table)
+                     VALUES (%s, %s, %s, %s, %s)
+                     RETURNING command_id"""
+            sql_params = [name, software.id, description, cli_cmd,
+                          parameters_table]
+            TRN.add(sql, sql_params)
+            c_id = TRN.execute_fetchlast()
+
+        return cls(c_id)
+
+    @property
+    def name(self):
+        """The name of the command
+
+        Returns
+        -------
+        str
+            The name of the command
+        """
+        with TRN:
+            sql = """SELECT name
+                     FROM qiita.software_command
+                     WHERE command_id = %s"""
+            TRN.add(sql, [self.id])
+            return TRN.execute_fetchlast()
+
+    @property
+    def description(self):
+        """The description of the command
+
+        Returns
+        -------
+        str
+            The description of the command
+        """
+        with TRN:
+            sql = """SELECT description
+                     FROM qiita.software_command
+                     WHERE command_id = %s"""
+            TRN.add(sql, [self.id])
+            return TRN.execute_fetchlast()
+
+    @property
+    def cli(self):
+        """The CLI used to call the command
+
+        Returns
+        -------
+        str
+            The CLI used to call the command
+        """
+        with TRN:
+            sql = """SELECT cli_cmd
+                     FROM qiita.software_command
+                     WHERE command_id = %s"""
+            TRN.add(sql, [self.id])
+            return TRN.execute_fetchlast()
+
+    @property
+    def parameters_table(self):
+        """The table in which the parameters of th e command are stored
+
+        Returns
+        -------
+        str
+            The name of the table
+        """
+        with TRN:
+            sql = """SELECT parameters_table
+                     FROM qiita.software_command
+                     WHERE command_id = %s"""
+            TRN.add(sql, [self.id])
+            return TRN.execute_fetchlast()
+
+
+class Software(QiitaObject):
+    r"""A software available in the system
```

Comment:
'A software' sounds rather strange, would it be more appropriate to say "A software package" ...